### ISSUE TITLE
Exception when i call a wrong URL

### ISSUE THREAD

**Issue by marcelalburg** [falconry/falcon#281]
Hello,

if i call the URL "/xxx.xxx?test=123&ord=%n" there is an Key Error Exception:

```python
From callback <function uwsgi_pypy_wsgi_handler at 0x0000000002edcbd8>:
Traceback (most recent call last):
  File "c callback", line 472, in uwsgi_pypy_wsgi_handler
  File "./falcon/api.py", line 88, in __call__
    req = self._request_type(env)
  File "./falcon/request.py", line 193, in __init__
    self.query_string = uri.decode(env['QUERY_STRING'])
  File "./falcon/util/uri.py", line 187, in decode
    char, byte = _HEX_TO_BYTE[token[:2]]
KeyError: 'n'
```

File: falcon/util/uri.py Line: 187
```python
       for token in tokens[1:]:
            char, byte = _HEX_TO_BYTE[token[:2]]
            decoded_uri += char + token[2:]

            if only_ascii:
                only_ascii = (byte <= 127)
```

I'll fixed that with:

```python
        for token in tokens[1:]:
            try:
                char, byte = _HEX_TO_BYTE[token[:2]]
            except KeyError:
                continue
            decoded_uri += char + token[2:]

            if only_ascii:
                only_ascii = (byte <= 127)
```

**Comment by lwcolton** [comment]
This is a bug.  The query string that you supplied has an invalid escape character (%n)  The query string decode function should raise an HttpBadRequest error if the query string is invalid.  I will submit a pull request with a fix shortly.

**Comment by philiptzou** [comment]
Perhaps related to #588 and I have already submit a pull request to fix it: https://github.com/falconry/falcon/pull/597

**Comment by kgriffs** [comment]
I just did a quick check and this seams to be fixed in 1.0.0rc2. Please reopen if you find this is not the case. Thanks!
